src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -475,9 +478,34 @@ def run_octave_calibration(self):
         for element in elements:
             self._qm.calibrate_element(element)
 
-    def set_parameter_of_bus(  # pylint: disable=too-many-locals
+    def get_controller_from_bus(self, bus: str) -> str | None:
```

Comment:
Renaming since it returns just the type, instead of the actual controller.

```suggestion
    def get_controller_type_from_bus(self, bus: str) -> str | None:
```